block/submit.go

```diff
@@ -26,6 +27,17 @@ func (m *Manager) SubmitLoop(ctx context.Context) {
 
 	// defer func to clear the channels to release blocked goroutines on shutdown
 	defer func() {
+		// recover from panic, emit health status event and return the error
```

Comment:
why u handle it in recover. instead of just avoiding the panic here
```
err := m.HandleSubmissionTrigger()
		if err != nil {
			panic(fmt.Errorf("handle submission trigger: %w", err))
		}
```

Reply:
I guess the idea is that it would be considered an unrecoverable error, so might as well panic. However, we would still like to get an event about it

Reply:
@zale144 why we need the event if the rpc server gonna be shut down anyway due to the panic?
don't we need `no panic + event` instead of the existing panic?